src/wasm/wasm-validator.cpp

```diff
@@ -2446,7 +2446,79 @@ void FunctionValidator::visitTry(Try* curr) {
 }
 
 void FunctionValidator::visitTryTable(TryTable* curr) {
-  // TODO
+  shouldBeTrue(
+    getModule()->features.hasExceptionHandling(),
+    curr,
+    "try_table requires exception-handling [--enable-exception-handling]");
+  if (curr->type != Type::unreachable) {
+    shouldBeSubType(curr->body->type,
+                    curr->type,
+                    curr->body,
+                    "try_table's type does not match try_table body's type");
+  } else {
+    shouldBeEqual(curr->body->type,
+                  Type(Type::unreachable),
+                  curr,
+                  "unreachable try_table must have unreachable try_table body");
+  }
+
+  shouldBeEqual(curr->catchTags.size(),
+                curr->catchDests.size(),
+                curr,
+                "the number of catch tags and catch destinations do not match");
+  shouldBeEqual(curr->catchTags.size(),
+                curr->catchRefs.size(),
+                curr,
+                "the number of catch tags and catch refs do not match");
+  shouldBeEqual(curr->catchTags.size(),
+                curr->sentTypes.size(),
+                curr,
+                "the number of catch tags and sent types do not match");
+
+  Type exnref = Type(HeapType::exn, Nullable);
+  for (Index i = 0; i < curr->catchTags.size(); i++) {
+    Name tagName = curr->catchTags[i];
+    if (!tagName) {
+      continue;
+    }
+
+    // Check tag validity
+    auto* tag = getModule()->getTagOrNull(tagName);
+    if (!shouldBeTrue(tag != nullptr, curr, "")) {
+      getStream() << "catch's tag name is invalid: " << tagName << "\n";
+    } else if (!shouldBeEqual(tag->sig.results, Type(Type::none), curr, "")) {
```

Comment:
```suggestion
    } else if (!shouldBeEqual(tag->sig.results, Type::none, curr, "")) {
```

Reply:
Won't compile, unfortunately, because `Type::none` is a `Type::BasicType`, not a `Type`, so the template gets confused :(

Reply:
Oh, too bad... thanks for the info.